### PR TITLE
refactor: simplify da type checking

### PR DESCRIPTION
**Pull request by bwanglzu**
`DocumentArray` is `MutableSequence` -> `Sequence` -> `Iterable`.

`iter(da)` is `Generator` -> `Iterator` -> `Iterable`.

`chain(da1, da2)` is both `Iterable` and `Iterator`.

Maybe we only need to check 1 type

```python
from docarray import DocumentArray, Document

from typing import Generator, Iterator, Sequence, Iterable

da = DocumentArray([Document(), Document(), Document()])

print(isinstance(da, DocumentArray))
print(isinstance(da, Sequence))
print(isinstance(da, Iterable))


my_iter = iter(da)

print(isinstance(my_iter, Iterator))
print(isinstance(my_iter, Generator))

from itertools import chain

da1 = DocumentArray([Document()])
da2 = DocumentArray([Document(), Document()])

# chaining da1 and da2 numbers
da = chain(da1, da2)
print(isinstance(da, Iterator))
print(isinstance(da, Iterable))
print(type(da))
```

### CODE REVIEW

**Comment by hanxiao** [comment]
@alaeddine-13 can you review it

File: docarray/array/chunk.py [hunk 1]
```diff
@@ -31,12 +28,7 @@ def __init__(self, docs, reference_doc: 'Document'):
         """
         self._ref_doc = reference_doc
         super().__init__(docs)
-        if (
-            isinstance(
-                docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-            )
-            and self._ref_doc is not None
-        ):
+        if isinstance(docs, (Iterable, Iterator)) and self._ref_doc is not None:
```
Comment (alaeddine-13) [inline docarray/array/chunk.py:31]: Iterator is also a subclass of Iterable so we can just keep Iterable here
```suggestion
        if isinstance(docs, Iterable) and self._ref_doc is not None:
```

File: docarray/array/match.py [hunk 2]
```diff
@@ -25,12 +22,7 @@ class MatchArray(DocumentArrayInMemory):
     def __init__(self, docs, reference_doc: 'Document'):
         self._ref_doc = reference_doc
         super().__init__(docs)
-        if (
-            isinstance(
-                docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-            )
-            and self._ref_doc is not None
-        ):
+        if isinstance(docs, (Iterable, Iterator)) and self._ref_doc is not None:
```
Comment (alaeddine-13) [inline docarray/array/match.py:25]:
```suggestion
        if isinstance(docs, Iterable) and self._ref_doc is not None:
```

File: docarray/array/storage/annlite/backend.py [hunk 3]
```diff
@@ -77,9 +75,7 @@ def _init_storage(
 
         self.clear()
 
-        if isinstance(
-            _docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-        ):
+        if isinstance(_docs, (Iterable, Iterator)):
```
Comment (alaeddine-13) [inline docarray/array/storage/annlite/backend.py:78]:
```suggestion
        if isinstance(_docs, Iterable):
```

File: docarray/array/storage/memory/backend.py [hunk 4]
```diff
@@ -36,7 +34,7 @@ def _init_storage(
             return
         elif isinstance(
             _docs,
-            (DocumentArray, Sequence, Generator, Iterator, itertools.chain),
+            (Iterable, Iterator),
```
Comment (alaeddine-13) [inline docarray/array/storage/memory/backend.py:37]: just Iterable

File: docarray/array/storage/qdrant/backend.py [hunk 5]
```diff
@@ -105,9 +103,7 @@ def _init_storage(
         # is provided, :class:`DocumentArraySqlite` will clear the existing
         # table and load the given `docs`
         self.clear()
-        if isinstance(
-            docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-        ):
+        if isinstance(docs, (Iterable, Iterator)):
```
Comment (alaeddine-13) [inline docarray/array/storage/qdrant/backend.py:106]:
```suggestion
        if isinstance(docs, Iterable):
```

File: docarray/array/storage/sqlite/backend.py [hunk 6]
```diff
@@ -116,9 +114,7 @@ def _init_storage(
 
         if _docs is None:
             return
-        elif isinstance(
-            _docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-        ):
+        elif isinstance(_docs, (Iterable, Iterator)):
```
Comment (alaeddine-13) [inline docarray/array/storage/sqlite/backend.py:117]:
```suggestion
        elif isinstance(_docs, Iterable):
```

File: docarray/array/storage/weaviate/backend.py [hunk 7]
```diff
@@ -90,9 +88,7 @@ def _init_storage(
         # table and load the given `docs`
         if _docs is None:
             return
-        elif isinstance(
-            _docs, (DocumentArray, Sequence, Generator, Iterator, itertools.chain)
-        ):
+        elif isinstance(_docs, (Iterable, Iterator)):
```
Comment (alaeddine-13) [inline docarray/array/storage/weaviate/backend.py:91]:
```suggestion
        elif isinstance(_docs, Iterable):
```

**Review by alaeddine-13 — COMMENTED** [review]
Maybe it's clearer to just check for Iterable

**Review by alaeddine-13 — APPROVED** [review]
LGTM

**Pull request merged**